addons/website_sale/views/templates.xml

```diff
@@ -950,9 +950,14 @@
 
     <!-- This template is the one at the "Review order" step (the first one) on the checkout workflow. -->
     <template id="cart_lines" name="Shopping Cart Lines">
-        <div t-if="not website_sale_order or not website_sale_order.website_order_line" class="js_cart_lines alert alert-info">
-          Your cart is empty!
-        </div>
+        <t t-if="not website_sale_order or not website_sale_order.website_order_line">
+            <div class="js_cart_lines alert alert-info">
+              Your cart is empty!
+            </div>
+            <div t-if="website_sale_order and website_sale_order.pending_coupon_id" class="js_cart_lines alert alert-warning">
+              The following promo code will be applied on checkout: <strong t-esc="website_sale_order.pending_coupon_id.code"/>
```

Comment:
This can't be in website_sale as coupon doesn't exists, you need to xpath from website_sale_coupon.